sound/soc/sdw_utils/soc_sdw_utils.c

```diff
@@ -1203,6 +1314,13 @@ int asoc_sdw_parse_sdw_endpoints(struct snd_soc_card *card,
 				    !(dai_info->quirk_exclude ^ !!(dai_info->quirk & ctx->mc_quirk)))
 					continue;
 
+				ret = is_sdca_function_present(dev, codec_info, adr_link, i, j);
+				if (ret < 0)
+					return ret;
+
+				if (!ret)
```

Comment:
is_sdca_function_present()
ret < 0 is error
ret == 0 is `true`
ret == 1 is `false`
?
I would wasp the true/false with a comment.

Reply:
or not, I'm not sure what is going on ;)

Reply:
My first thought wat let is_sdca_function_present() return true/false only and it will be Intuitive. However, I would like to include the error checking in is_sdca_function_present(), so I changed the return type to int. Maybe we should rename is_sdca_function_present() as it is not just checking is SDCA function present. The purpose of the helper is to decide whether we should skip the endpoint. We should only skip the endpoint if there are more than 1 endpoints and we can get the SDCA supported functions and none of the supported functions matches the DAI type.